### ISSUE TITLE
configure ignores --with-libdir

### ISSUE THREAD

**Issue by pacohope** [eduardok/libsmbclient-php#37]
I am compiling on FreeBSD 10.2-RELEASE. I've installed the samba-libsmbclient port, (which is based on samba 3.6, but whatever). 

I am running:
`./configure --prefix=/usr/local --with-libdir=/usr/local/lib`
or
`./configure --prefix=/usr/local`

In both cases, configure dies with this error:
```configure:4427: cc -o conftest -g -O2   -lsmbclient conftest.c -lsmbclient   >&5
/usr/bin/ld: cannot find -lsmbclient```

I set the environment variable `LDFLAGS=-L/usr/local/lib` and then I see
```configure:6283: cc -o conftest -g -O2  -L/usr/local/lib conftest.c conftstm.o >&5```

and configure works correctly. Seems like the configure script isn't really processing many of the args it says it does.

**Comment by aklomp** [comment]
`--with-libdir` doesn't set the full path to the library, it only changes the libdir's value from the default of `lib` to something user-defined. Example, if your system stores libraries in /usr/lib64, you'd use `--prefix=/usr --with-libdir=lib64`.

To look for libsmbclient in `/usr/local/lib`, use this syntax:

```sh
./configure --with-libsmbclient=/usr/local
```
The configure script will expand the base directory of `/usr/local` and look for the shared library inside `/usr/local/$LIBDIR` (where `$LIBDIR` is user-settable as per above, but defaults to `lib`), and for the header file inside `/usr/local/include`.

(Disclaimer, I reverse-engineered this from the code, I haven't tested that it actually works.)

**Comment by aklomp** [comment]
Hi, may I close this issue?

**Comment by pacohope** [comment]
I am so sorry. Yes, of course you were right. Please close the issue. That was the correct option. I posted on the ownCloud forums where it might do someone else some good: https://forum.owncloud.org/viewtopic.php?f=36&t=32578

As far as I can tell, none of this stuff works except on Samba 3.6. Is there some plan to make it work with a newer version of Samba, or is there some fundamental stuff that has changed that makes this impossible to port to the new versions?

Paco

**Comment by remicollet** [comment]
It works perfectly (test suite succeeds) with samba 4.3 (on Linux)

**Comment by aklomp** [comment]
I know this is not an OwnCloud support channel, but are you sure you also need the `smbclient` binary if you've also installed our PHP extension? As I understand it, OwnCloud uses the `smbclient` binary only as a fallback if our extension is not available. Chances are you're not running our extension, but are relying on the `smbclient` workaround, and maybe that's why things fall apart on Samba 4.

**Comment by remicollet** [comment]
> As I understand it, OwnCloud uses the smbclient binary only as a fallback if our extension is not available.

Yes but no ;)

Owncloud use the smbclient command to retrieve a list of users in another part of the code. The extension is only used via icewind/SMB library for file access.

**Comment by pacohope** [comment]
You know, I’m pretty sure you’re right. I think I have built your extension correctly (where “correctly” means compiles and installs with no errors). I’m fairly sure I’ve got it into the PHP environment correctly. (Perhaps I should look more carefully)  But I’m also certain ownCloud is relying on the smbclient binary (and I think that’s a known ownCloud limitation related to passing the OC session credentials to the samba server on a per-user basis).

The other reason I went with Samba 3.6 was because I could not get your extension to compile under FreeBSD with Samba 4.3. Part of that, I think, is related to FreeBSD’s package putting things in weird places. Like libsmbclient.h ended up in /usr/local/include/samba4 instead of just /usr/local/include. I wrestled with autoconf a while, but couldn’t sort it out. Since rolling back to 3.6 worked, I did that and quit.

If you’d like, I might be able to get a clean environment and try a bit harder to figure out what’s going wrong compiling on FreeBSD against Samba 4.3.

So basically: you’re right that I’m relying on the smbclient binary, and that’s an ownCloud thing, not a php-smbclient thing. Unrelated to that, I think your autoconf isn’t very flexible if the OS puts stuff in places you don’t expect. I’d be happy to help troubleshoot that if it would be of value to you.

Thanks for helping me out, even when it’s not really related to your code.

Paco

**Comment by remicollet** [comment]
> ike libsmbclient.h ended up in /usr/local/include/samba4

The search is

       for i in /usr/local/include /usr/local/include/samba-* /usr/include /usr/include/samba-* ; do

Perhaps should be 

       for i in /usr/local/include /usr/local/include/samba* /usr/include /usr/include/samba* ; do

Notice: the real good solution is to rely on pkgconfig output, and get rid of this check.

**Comment by eduardok** [comment]
Closing it for the time being, given original issue.
